test/powershell/Modules/Microsoft.PowerShell.Utility/FormatOutOfBandData.Tests.ps1

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe 'Formatting out of band data tests' -Tags 'CI' {
+
+    Context 'Outputting out of band data renders properly' {
+        
+        BeforeAll {
+            $formatCommandTestCases = @(
+                @{
+                    FormatCommand = 'Format-Table'
+                    DataType      = 'TableRowEntry'
+                },
+                @{
+                    FormatCommand = 'Format-List'
+                    DataType      = 'ListViewEntry'
+                },
+                @{
+                    FormatCommand = 'Format-Wide'
+                    DataType      = 'WideViewEntry'
+                },
+                @{
+                    FormatCommand = 'Format-Custom'
+                    DataType      = 'ComplexViewEntry'
+                }
+            )
+            $oobTypes = 'Error', 'Warning', 'Verbose', 'Debug', 'Information'
+            $formatCommandWithOobTypeTestCases = foreach ($oobType in $oobTypes) {
+                foreach ($item in $formatCommandTestCases) {
+                    @{
+                        OobType       = $oobType
+                        FormatCommand = $item.FormatCommand
+                        DataType      = $item.DataType
+                    }
+                }
+            }
+
+            function Test-OutOfBandFormatting {
```

Comment:
Could you please add a comment that is expected output of the function?

Reply:
Done.